apps/client/src/features/app-settings/panel/project-panel/ProjectList.tsx

```diff
@@ -1,19 +1,42 @@
-import { IconButton, Menu, MenuButton, MenuItem, MenuList } from '@chakra-ui/react';
-import { IoEllipsisHorizontal } from '@react-icons/all-files/io5/IoEllipsisHorizontal';
+import { useMemo, useState } from 'react';
 
 import { useProjectList } from '../../../../common/hooks-query/useProjectList';
 import * as Panel from '../PanelUtils';
 
-import style from './ProjectPanel.module.scss';
+import ProjectListItem from './ProjectListItem';
+
+export type EditMode = 'rename' | 'duplicate' | null;
 
 export default function ProjectList() {
-  const { data } = useProjectList();
+  const { data, refetch } = useProjectList();
   const { files, lastLoadedProject } = data;
 
-  // extract currently loaded from file list
-  const currentlyLoadedIndex = files.findIndex((project) => project.filename === lastLoadedProject);
-  const projectFiles = [...files];
-  const current = projectFiles.splice(currentlyLoadedIndex, 1)[0];
+  const [editingMode, setEditingMode] = useState<EditMode | null>(null);
+  const [editingFilename, setEditingFilename] = useState<string | null>(null);
+
+  const handleToggleEditMode = (editMode: EditMode, filename: string | null) => {
+    setEditingMode((prev) => (prev === editMode && filename === editingFilename ? null : editMode));
+    setEditingFilename(filename);
+  };
```

Comment:
it seems that in `<ProjectListItem>` we short circuit submits where `filename` is falsy, would that not mean that we would never get a `null` here. Is the same not true for `EditMode`?

Reply:
when we cancel, we set `editingFilename` to `null`, same for `editMode` is `null` if we're not editing anything